test/verify/check-login

```diff
@@ -561,5 +561,128 @@ account    required     pam_succeed_if.so user ingroup %s""" % m.get_admin_group
+
+        # add it to the pam config
+        if module == 'pam_tally2':
+            self.sed_file("/common-auth/ { iauth required pam_tally2.so deny=4\n }", "/etc/pam.d/cockpit")
+        else:
+            # see https://access.redhat.com/solutions/62949
+            self.sed_file("""/pam_unix/ {
+                      s/sufficient/[success=1 default=ignore]/\n
+                      aauth [default=die] pam_faillock.so authfail audit deny=4 unlock_time=600\n
+                      aauth sufficient pam_faillock.so authsucc audit deny=4 unlock_time=600\n
+                      }""", "/etc/pam.d/password-auth")
+
+        m.execute("grep -r '{}' /etc/pam.d".format(module))
+
+        m.start_cockpit()
+
+        # pam_faillock tries to chown the file to be owned by the user itself,
+        # which is currently an selinux fail on the following distros:
+        if m.image in ['centos-8-stream', 'fedora-31', 'fedora-32', 'rhel-8-2', 'rhel-8-2-distropkg', 'rhel-8-3']:
+            # https://bugzilla.redhat.com/show_bug.cgi?id=1836182
+            self.allow_journal_messages('.*type=1400.*avc:  denied  { chown }.*comm="cockpit-session".*')
+
+        if module == 'pam_tally2':
+            # at first, we should have no log file
+            m.execute("! test -f {}".format(logfile('admin')))
+
+        # make sure we can still login
+        b.login_and_go("/system")
+        b.logout()
+
+        if module == 'pam_tally2':
+            # now we should have it
+            m.execute("test -f {}".format(logfile('admin')))
+
+        # and it should show zero fails
+        expect_fail_count(0, must_exist=False)
+
+        # try three bogus login attempts
+        for n in [1, 2, 3]:
+            expect_failed_login("admin", "bad", n)
+        expect_fail_count(3)
+
+        # make sure we can still login
+        expect_successful_login("admin", "foobar")
+
+        # after the success, try three more bogus login attempts
+        for n in [1, 2, 3]:
+            expect_failed_login("admin", "bad", n)
+        expect_successful_login("admin", "foobar")
+
+        # now try four, which should lock the account
+        for n in [1, 2, 3, 4]:
+            expect_failed_login("admin", "bad", n)
+
+        # logging in should fail now
+        if module == 'pam_tally2': # this is counted differently by each module
+            n += 1
+        expect_failed_login("admin", "foobar", n)
+
+        # having given the correct password the last time should not have helped things
+        if module == 'pam_tally2':
+            n += 1
+        expect_failed_login("admin", "foobar", n)
+
+        # but we can reset the lockout
+        m.execute(module + " --reset --user admin")
+        expect_fail_count(0)
+
+        # and login again
+        expect_successful_login("admin", "foobar")
+
+        # fedora-coreos logs in via sshd, which forbids root logging in with a password
+        if m.image not in ['fedora-coreos']:
```

Comment:
I think you asked at some point to change this to conditionalising on some flag, rather than the image name?

Reply:
That would be `m.ostree_image`.

Reply:
Right, but as I wrote above I changed my mind -- it's not obviously specific to ostree (like e. g. "can remove an rpm or write to /usr"), but specific to FCOS.

Reply:
Right.  Makes sense.  +1 from me.